src/renderer/views/swap/SwapView.tsx

```diff
@@ -138,49 +128,96 @@ const SuccessRouteView: React.FC<Props> = ({
     keystoreService: { keystoreState$, validatePassword$ }
   } = useWalletContext()
 
+  const { chain: sourceChain } = sourceAsset.type === AssetType.SYNTH ? mayaDetails.asset : sourceAsset
+  const { chain: targetChain } = targetAsset.type === AssetType.SYNTH ? mayaDetails.asset : targetAsset
+
+  const poolAddressThor$ = poolAddressesByChain$(sourceChain)
+  const poolAddressMaya$ = poolAddressesByChainMaya$(sourceChain)
+  const combinedPoolAddresses = useObservableState(
```

Comment:
Please double-check the logic here, this is causing an infinite re-render

Reply:
ok